core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -802,6 +808,12 @@ object KafkaConfig {
   val ShareGroupRecordLockPartitionLimitDoc = "Share-group record lock limit per share-partition."
   val ShareGroupMaxGroupsDoc = "The maximum number of share groups."
   val ShareGroupMaxSizeDoc = "The maximum number of consumers that a single share group can accommodate."
+  val ShareGroupSessionTimeoutMsDoc = "The timeout to detect client failures when using the share group protocol."
+  val ShareGroupMinSessionTimeoutMsDoc = "The minimum allowed session timeout."
+  val ShareGroupMaxSessionTimeoutMsDoc = "The maximum allowed session timeout."
+  val ShareGroupHeartbeatIntervalMsDoc = "The heartbeat interval given to the members of a share group."
+  val ShareGroupMinHeartbeatIntervalMsDoc = "The minimum heartbeat interval."
+  val ShareGroupMaxHeartbeatIntervalMsDoc = "The maximum heartbeat interval."
 
```

Comment:
@AndrewJSchofield Should we have the elaborated description, may be need to update the KIP as well? i.e. `The minimum heartbeat interval for registered share group members.`

Reply:
Good point. I reckon "for share group members" would work.

Reply:
This file has actually been converted into Java in trunk now (or in the process of merging, anyway). There's some merge work building up out there.

Reply:
Thanks for your comments @apoorvmittal10 @AndrewJSchofield . I have updated the description for `ShareGroupMinHeartbeatIntervalMsDoc`, `ShareGroupMaxHeartbeatIntervalMsDoc`, `ShareGroupMinSessionTimeoutMsDoc` and `ShareGroupMaxSessionTimeoutMsDoc`.